test/CMakeLists.txt

```diff
@@ -222,4 +228,164 @@ foreach (test_src IN LISTS test_srcs)
   endforeach() # CUB targets
 endforeach() # Source file
 
+
+# Catch2 tests
+find_package(CUDAToolkit REQUIRED)
+foreach(cub_target IN LISTS CUB_TARGETS)
+  cub_get_target_property(config_prefix ${cub_target} PREFIX)
+
+  # The actual name of the test's target:
+  set(target ${config_prefix}.c2h)
+
+  add_library(${target} SHARED
+    catch2_test_helper.h 
+    c2h/custom_type.cuh 
+    c2h/generators.cuh 
+    c2h/generators.cu)
+
+  target_link_libraries(${target} PRIVATE CUDA::curand)
+
+  set_property(TARGET ${target} PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+  target_link_options(${target} PRIVATE "-shared")
+
+  if ("NVCXX" STREQUAL "${CMAKE_CUDA_COMPILER_ID}")
+    target_link_options(${target} PRIVATE "-cuda")
+    target_compile_options(${target} PRIVATE "-fPIC")
+  endif()
+
+  target_include_directories(${target} 
+    PUBLIC "${CUB_SOURCE_DIR}/test")
+
+  cub_clone_target_properties(${target} ${cub_target})
+endforeach()
+
+function(cub_add_catch2_test src name defs target config_prefix)
```

Comment:
Add a docstring similar to `cub_add_test`. We should try to make the signatures match if possible.